packages/web-client/app/models/workflow/workflow-session.ts

```diff
@@ -1,14 +1,37 @@
+import WorkflowPersistence from '@cardstack/web-client/services/workflow-persistence';
 import { tracked } from '@glimmer/tracking';
 
 export interface ArbitraryDictionary {
   [key: string]: any;
 }
+
 export default class WorkflowSession {
+  workflow: any;
+  workflowPersistenceStorage: WorkflowPersistence | undefined;
+
+  constructor(workflow?: any) {
+    this.workflow = workflow;
+
+    // Allow WorkflowSession to be instantiated as a POJO (for unit tests)
+    if (this.workflow && typeof this.workflow.owner.lookup === 'function') {
+      this.workflowPersistenceStorage = this.workflow.owner.lookup(
+        'service:workflow-persistence'
+      );
```

Comment:
Also, as I think I commented elsewhere, it would be better for the Workflow class to be responsible for looking up/injecting the service, and then have this class just read it off the workflow instance.

Reply:
Yeah that makes total sense. Will adjust it.